cmd/ssh/ssh.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/sourcepods/sourcepods/cmd"
+	"github.com/sourcepods/sourcepods/pkg/gitssh"
+	"github.com/sourcepods/sourcepods/pkg/storage"
+	jaeger "github.com/uber/jaeger-client-go"
+	"github.com/uber/jaeger-client-go/config"
+	"github.com/urfave/cli"
+)
+
+type sshConf struct {
+	SSHAddr        string
+	HostKeyPath    string
+	StorageGRPCURL string
+	HTTPAddr       string
+	LogJSON        bool
+	LogLevel       string
+	TracingURL     string
+}
+
+var (
+	sshConfig = sshConf{}
+	sshFlags  = []cli.Flag{
+		cli.StringFlag{
+			Name:        cmd.FlagSSHAddr,
+			EnvVar:      cmd.EnvSSHAddr,
+			Value:       ":22",
+			Usage:       "The SSH address to listen on",
+			Destination: &sshConfig.SSHAddr,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagSSHHostKeyPath,
+			EnvVar:      cmd.EnvSSHHostKeyPath,
+			Value:       "/etc/ssh/",
+			Usage:       "The path to looks for ssh host-keys in",
+			Destination: &sshConfig.HostKeyPath,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagStorageGRPCURL,
+			EnvVar:      cmd.EnvStorageGRPCURL,
+			Usage:       "The storage's grpc url to connect with",
+			Destination: &sshConfig.StorageGRPCURL,
+		},
+		cli.BoolFlag{
+			Name:        cmd.FlagLogJSON,
+			EnvVar:      cmd.EnvLogJSON,
+			Usage:       "The logger will log json lines",
+			Destination: &sshConfig.LogJSON,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagLogLevel,
+			EnvVar:      cmd.EnvLogLevel,
+			Usage:       "The log level to filter logs with before printing",
+			Value:       "info",
+			Destination: &sshConfig.LogLevel,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagHTTPAddr,
+			EnvVar:      cmd.EnvHTTPAddr,
+			Usage:       "The address SourcePods runs a http server on",
+			Value:       ":3021",
+			Destination: &sshConfig.HTTPAddr,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagTracingURL,
+			EnvVar:      cmd.EnvTracingURL,
+			Usage:       "The url to send spans for tracing to",
+			Destination: &sshConfig.TracingURL,
+		},
+	}
+)
+
+func sshAction(c *cli.Context) error {
+	if sshConfig.StorageGRPCURL == "" {
+		return errors.New("the storage grpc url can not be empty")
+	}
+	if sshConfig.HTTPAddr == "" {
```

Comment:
This is for connecting to the API-service, so we can do name=>uuid-mapping. But since I'm leaving that until next PR I don't mind removing this all together :thinking: